lib/ansible/modules/files/xml.py

```diff
@@ -545,8 +545,11 @@ def set_target_inner(module, tree, xpath, namespaces, attribute, value):
         if not is_node(tree, xpath, namespaces):
             changed = check_or_make_target(module, tree, xpath, namespaces)
     except Exception as e:
-        module.fail_json(msg="Xpath %s causes a failure: %s\n  -- tree is %s" %
-                             (xpath, e, etree.tostring(tree, pretty_print=True)), exception=traceback.format_exc(e))
+        missing_namespace = ""
+        if len(tree.getroot().nsmap) > 0 and ":" not in xpath:
+            missing_namespace = "XML document has namespace(s) defined, but no namespace prefix(es) used in xpath!\n"
+        module.fail_json(msg="%sXpath %s causes a failure: %s\n  -- tree is %s" %
```

Comment:
I am not sure if `%sXpath %s` is intentional. Seems a bit weird.
Everything else appears fine.

Maybe we should add a TODO or NOTE comment specifying the lack of namespace checks for child elements.

Reply:
Yes, it seems a bit weird, but couldn't think of better way to include an optional string ending with a new line before the existing one. Didn't want to mix concatenation and substitution. Any suggestion is welcome.

Reply:
Fine by me. I would still add that note related to child-namespaces so people skimming the code understand the limitations and/or can contribute a more comprehensive fix for this.